src/encoded/static/libs/ui/modal.js

```diff
@@ -286,53 +290,37 @@ ModalFooter.defaultProps = {
 /**
  * Renders the modal without conditions -- conditions need to be handled outside this component.
  */
-export class ModalElement extends React.Component {
-    constructor(props) {
-        super(props);
-        this.handleTab = this.handleTab.bind(this);
-        this.modalEl = document.createElement('div');
-        this.focusableElements = null;
-    }
+const ModalElement = ({
+    modalChildren,
+    descriptionId,
+    addClasses,
+    focusId,
+    labelId,
+}) => {
+    const modalEl = React.useRef(null);
+    const focusableElements = React.useRef(null);
 
-    componentDidMount() {
-        // Modal HTML gets injected at the end of <body> so the backdrop overlay works, even with other
-        // fixed elements on the page. The "modal-root" div is defined in app.js at the end of the
-        // <body> section.
-        const modalRoot = document.getElementById('modal-root');
-        modalRoot.appendChild(this.modalEl);
-
-        // Focus screen reader on the given element if specified.
-        if (this.props.focusId) {
-            const focusElement = document.getElementById(this.props.focusId);
-            if (focusElement) {
-                focusElement.focus();
-            }
+    /**
+     * Retrieve the current <div> DOM node for the modal. Lazily create this node if needed.
+     * @returns {DOM node} Injected DOM node reference for modal
+     */
+    const getModalEl = () => {
+        if (modalEl.current === null) {
```

Comment:
null or falsey?

Reply:
Actually, not sure. I’ll leave it for now since it’s existing code.